apps/andi/test/unit/andi_web/live/dataset_live_view_test.exs

```diff
@@ -0,0 +1,176 @@
+defmodule AndiWeb.DatasetLiveViewTest do
+  use AndiWeb.ConnCase
+  use Phoenix.ConnTest
+  import Phoenix.LiveViewTest
+
+  alias Andi.DatasetCache
+
+  alias SmartCity.TestDataGenerator, as: TDG
+
+  @endpoint AndiWeb.Endpoint
+  @url_path "/datasets/live"
+
+  setup do
+    GenServer.call(DatasetCache, :reset)
+  end
+
+  describe "Basic live page load" do
+    test "loads all datasets", %{conn: conn} do
+      datasets = Enum.map(1..3, fn _x -> TDG.create_dataset([]) end)
+
+      DatasetCache.put_datasets(datasets)
+
+      assert {:ok, _view, html} = live(conn, @url_path)
+
+      table_text = floki_get_text(html, ".datasets-index__table")
+
+      Enum.each(datasets, fn dataset ->
+        assert table_text =~ dataset.business.dataTitle
+      end)
+    end
+
+    test "shows No Datasets when there are no rows to show", %{conn: conn} do
+      assert {:ok, view, html} = live(conn, @url_path)
+
+      assert html =~ "All Datasets"
```

Comment:
*This assertion* in *this test* confuses me.

Reply:
Me too.

We keep that "All Datasets" header up above the table at all times.

This test was initially just verifying that the live view loads and we didn't take it out.

Reply:
I updated the assertions to be more specific.